// xdl/build/Config.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getenv_1 = __importDefault(require("getenv"));
const internal_1 = require("./internal");
function getAPI() {
    if (internal_1.Env.isLocal()) {
        return {
            scheme: 'http',
            host: 'localhost',
            port: 3000,
        };
    }
    else if (internal_1.Env.isStaging()) {
        return {
            scheme: getenv_1.default.string('XDL_SCHEME', 'https'),
            host: 'staging.exp.host',
            port: getenv_1.default.int('XDL_PORT', 0) || null,
        };
    }
    else {
        return {
            scheme: getenv_1.default.string('XDL_SCHEME', 'https'),
            host: getenv_1.default.string('XDL_HOST', 'exp.host'),
            port: getenv_1.default.int('XDL_PORT', 0) || null,
        };
    }
}
const config = {
    api: getAPI(),
    developerTool: 'expo-cli',
};
exports.default = config;
